### ISSUE TITLE
DiscordBot Provider?

### ISSUE THREAD

**Issue by csharpfritz** [FritzAndFriends/TagzApp#37]
It was suggested that we also provide a DiscordBot implementation that watches a specific chatroom on a server and surfaces questions from that chatroom

**Comment by csharpfritz** [comment]
This will be very cool to support for the average live-streamer on YouTube / Twitch with a Discord server.

Also:  for events, this will provide a very interactive opportunity for viewers to participate with the myriad of Discord features that we could enable in a bot.

**Comment by failtwoconnect** [comment]
I will do this.

**Comment by csharpfritz** [comment]
Thank you @failtwoconnect !  I've assigned you this issue

**Comment by failtwoconnect** [comment]
Only one bot for discord and passing in Oauth URL for adding to different servers.

**Comment by csharpfritz** [comment]
Yes, let's set up a similar proxy server that relays messages from
Discord.  An Azure Functions app might not be appropriate for this... we
will want to design this piece of architecture


Jeff

**Comment by failtwoconnect** [comment]
adding some documentation for oauth2 tokenization that we may want to implement. 

https://discord.com/oauth2/authorize | base URL
https://discord.com/api/oauth2/token | Token URL
https://discord.com/api/oauth2/token/revoke | Token Revocation URL

**Comment by csharpfritz** [comment]
Great idea to add Discord Auth!

**Comment by csharpfritz** [comment]
Following up here.. do you have any code for this?  Otherwise, I'll start on the Discord provider

**Comment by failtwoconnect** [comment]
Sorry I got busy with doing class work because I am going back to school. if you want to give it a shot. go for it. the documentation for it has gotten better from the last time I looked at it. 

There are some libraries that you could use for it DSharp or something like that will give you an API to use for the app. 

The permissions of the application need to be defined in the developer page from discord in order for the app to work. 

I will see if I can pop on to the stream tomorrow if you are doing it for the extra help.

sorry I couldn't add anything to the app.
